### DISCUSSION TITLE
Require that the matter reference field is 6 characters

### DISCUSSION THREAD

**Discussion by hagen31** [sadieblue/agilityblue.workspace-scripts#12] [Q&A] [answered]
Hello,

As this is my first script, I took the publicly available https://github.com/sadieblue/agilityblue.workspace-scripts/blob/master/Require%20that%20the%20matter%20reference%20field%20is%206%20characters%20long.ps1 and made two changes.

1. Changed to 10 characters
2. Changed the message to display

Instead of getting a custom message the error I received was "The results of a script have prevented this request to continue".

What am I missing?

I also what to force character convention is that possible?

Thank you,

Austin

**Comment by bmward80** [comment, marked as answer]
Hi @hagen31,

The comments are not particularly clear in the example script about how the error messages work, but the message you return when preventing a save is is only available within the output of your script, which can be viewed when you're looking at the execution results tab of the script. Currently, the error that the end users see when a script prevents an object from being saved is "The results of a script have prevented this request to continue". We're actively working on allowing custom error messages to be passed onto the user to be included in an update, but what you're seeing is by design as of now.

I hope this helps clear that up,
Brandon

**Comment by hagen31** [comment]
Alright looking forward to the custom message as the default is not descriptive of what rule is preventing as there could be multiple.

Any thoughts on being able to force character convention? IE four alphas and underscore and then six numeric?

**Comment by bmward80** [comment]
Yes, you would need to build out a regular expression that you could use to match. This post here shows how to build something similar that you could use as a starting point: [https://github.com/sadieblue/agilityblue.workspace-scripts/discussions/5](https://github.com/sadieblue/agilityblue.workspace-scripts/discussions/5)

There are a variety of online tools or AI that could help you with building out more specific patterns you could probably find easily enough but if you get stuck, let me know and I'll see if we can track something down for you.

Brandon

**Reply by hagen31** [reply]
Thank you I'll look at that as a starting point. I've been having no success with AI on another custom request so will keep digging.
